server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/EventDeadLettersIntegrationTest.java

```diff
@@ -0,0 +1,473 @@
+    private RetryEventsListener retryEventsListener = new RetryEventsListener();
+
+    @ClassRule
+    public static DockerCassandraRule cassandra = new DockerCassandraRule();
+
+    @Rule
+    public CassandraJmapTestRule cassandraJmapTestRule = CassandraJmapTestRule.defaultTestRule();
+
+    private GuiceJamesServer guiceJamesServer;
+    private MailboxProbeImpl mailboxProbe;
+
+    @Before
+    public void setUp() throws Exception {
+        retryEventsListener.clear();
+        guiceJamesServer = cassandraJmapTestRule.jmapServer(cassandra.getModule())
+            .overrideWith(new WebAdminConfigurationModule())
+            .overrideWith(binder -> Multibinder.newSetBinder(binder, MailboxListener.GroupMailboxListener.class).addBinding().toInstance(retryEventsListener));
+        guiceJamesServer.start();
+
+        DataProbe dataProbe = guiceJamesServer.getProbe(DataProbeImpl.class);
+        mailboxProbe = guiceJamesServer.getProbe(MailboxProbeImpl.class);
+        WebAdminGuiceProbe webAdminGuiceProbe = guiceJamesServer.getProbe(WebAdminGuiceProbe.class);
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminGuiceProbe.getWebAdminPort())
+            .build();
+
+        dataProbe.addDomain(DOMAIN);
+        dataProbe.addUser(BOB, BOB_PASSWORD);
+    }
+
+    @After
+    public void tearDown() {
+        guiceJamesServer.stop();
+    }
+
+    private void generateInitialEvent() {
+        mailboxProbe.createMailbox(MailboxPath.forUser(BOB, DefaultMailboxes.INBOX));
+    }
+
+    private void generateSecondEvent() {
+        mailboxProbe.createMailbox(MailboxPath.forUser(BOB, DefaultMailboxes.OUTBOX));
+    }
+
+    private String retrieveFirstFailedEventId() {
+        ArrayList<String> response = when()
+            .get(EventDeadLettersRoutes.BASE_PATH + "/groups/" + SERIALIZED_GROUP)
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .contentType(ContentType.JSON)
+            .extract()
+            .as(ArrayList.class);
+
+        return response.get(0);
+    }
+
+    @Test
+    public void failedEventShouldBeStoredInRetryListenerFailedEventsList() {
+        retryEventsListener.setRetriesBeforeSuccess(4);
+        generateInitialEvent();
+
+        when()
+            .get(EventDeadLettersRoutes.BASE_PATH + "/groups/" + SERIALIZED_GROUP)
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .contentType(ContentType.JSON)
+            .body(".", hasSize(1));
+    }
+
+    @Test
+    public void successfulEventShouldNotBeStoredInRetryListenerFailedEventsList() {
+        retryEventsListener.setRetriesBeforeSuccess(3);
+        generateInitialEvent();
+
+        when()
+            .get(EventDeadLettersRoutes.BASE_PATH + "/groups")
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .contentType(ContentType.JSON)
+            .body(".", hasSize(0));
+    }
+
+    @Test
+    public void failedEventGroupShouldBeStoredInDeadLetter() {
```

Comment:
I don't understand the test name, what is "failedEventGroup" ?

Reply:
Is `groupIdOfFailedEventShouldBeStoredInDeadLetter` better?

Reply:
:+1: